src/main/java/duke/task/EventTask.java

```diff
@@ -0,0 +1,83 @@
+package duke.task;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+
+import duke.Duke;
+
+/**
+ * An event task.
+ */
+public final class EventTask extends Task {
+    private LocalDate dateStart;
+    private LocalDate dateEnd;
+
+    /**
+     * Creates an event task.
+     * @param task The input string.
+     * @throws Duke.WrongFormatException If the input string is in the wrong format.
+     */
+    public EventTask(String task) throws Duke.WrongFormatException {
+        String description = getDescription(task);
+        if (description == null) {
+            throw new Duke.WrongFormatException("Whopsie daisies! I don't understand that format!");
+        }
+        this.description = description;
+    }
+
+    /**
+     * Creates an event task.
+     * @param isDone Whether the task is done.
+     * @param description The description of the task.
+     * @param dateStart The start date of the event.
+     * @param dateEnd The end date of the event.
+     */
+    public EventTask(boolean isDone, String description, String dateStart, String dateEnd) {
+        this.isDone = isDone;
+        this.description = description;
+        this.dateStart = LocalDate.parse(dateStart);
+        this.dateEnd = LocalDate.parse(dateEnd);
+    }
+
+    @Override
+    protected String getTaskTypeString() {
+        return squareBracketWrapper("E");
+    }
+
+    @Override
+    protected String saveToFileString() {
```

Comment:
This method sounds like it saves to the file, when it just returns the string in a save format. Maybe a name like toSaveString or toSaveFormat would be better?